ansible/playbooks/adhoc/docker_storage_to_overlay2/docker_storage_to_overlay2.yml

```diff
@@ -0,0 +1,125 @@
+# This playbook wraps the similarly named role in openshift-ansible.
+#
+# It's idempotent and safe to run repeatedly.  If the node's storage driver
+# is already overlay2, the ephemeral storage is left untouched.
+#
+# The role itself wipes all ephemeral storage with "atomic storage reset", so
+# this playbook drains nodes before invoking the role.
+#
+# Usage example:
+#
+#  ansible-playbook docker_storage_to_overlay2.yml -e cli_clusterid=testcluster
+#
+
+- hosts: localhost
+  gather_facts: no
+  become: no
+  user: root
+
+  tasks:
+
+  - name: Check for required variables
+    fail:
+      msg: "Please define {{ item }}"
+    when: "{{ item }} is undefined"
+    with_items:
+    - cli_clusterid
+    run_once: True
+
+##############################################
+# Master upgrade
+##############################################
+
+- hosts: "oo_clusterid_{{ cli_clusterid }}:&oo_hosttype_master"
+  gather_facts: yes  # docker_storage_to_overlay2 needs ansible_distribution
+  become: no
+  user: root
+  serial: 1
+
+  roles:
+  - role: ../../../../openshift/installer/atomic-openshift-3.7/roles/docker_storage_driver
+
```

Comment:
We're going to run this playbook on hosts that are running OCP 3.6 and 3.5 too. Is this role identical in `atomic-openshift-3.7`, `atomic-openshift-3.6` and `atomic-openshift-3.5`?

Reply:
Well, it's not in *any* of them yet.  ATM, I have `atomic-openshift-3.7` symlinked to my local `openshift-ansible` repo with my topic branch checked out.

This is the part I guess I'm still confused about how it's supposed to work.

* To begin with, how does one invoke a product-provided role from an `openshift-tools` playbook?

* And if we're invoking a new role that's supposed to come from the product, how do we deal with older versions of the product which lack these roles?  A local copy in `openshift-tools` is the only thing I can think of.

Reply:
Usually we request the role to be backported into all the versions we need. And then we call the role using the version of the installer that matches the version of OCP on a running cluster. So for most of Dedicated, we would call `../../../../openshift/installer/atomic-openshift-3.5/roles/docker_storage_driver` since Dedicated is currently running 3.5. Though since this is a one-time script, we might be able to leave the version hard-coded to 3.7. But we can only do that if we're sure there wont be different versions of this role for each version of OCP.

Once the role is present upstream (in the openshift-ansible repo), we'll vendor the appropriate version(s) into openshift-tools.